### PR TITLE
Development

### PR DESCRIPTION
**Pull request by SoleCincis**
add button functionality

### CODE REVIEW

File: screens/HomeScreen.js [hunk 1]
```diff
@@ -23,13 +25,39 @@ export default function HomeScreen() {
...
   let totalExpense = _sumBy(ALL_EXPENSES, 'amount');
 
+  const handleSubmitForm = () => {
+    //check whether the name is not empty and the amount is not negative
+    if (name !== '' && amount > 0) {
+      // single expense object
+      const expense = { name, amount };
+      // do not override previous values in the array
+      // use spread operator to access previous values
+      setExpenses([...expenses, expense]); // !!! INVALID ATTEMPT TO SPREAD NON-ITERABLE INSTANCE.
```
Comment (donni106) [inline screens/HomeScreen.js:35]: Is the issue #5 connected to this?

Comment (SoleCincis) [reply screens/HomeScreen.js:35]: it will next time, I can't find a way to edit the commit at this point .

Comment (donni106) [reply screens/HomeScreen.js:35]: Hm do not understand. Ok than show the problem in #5 more clear?

File: screens/HomeScreen.js [hunk 2]
```diff
@@ -2,39 +2,31 @@ import React, { useState } from 'react';
...
+const ALL_EXPENSES = [
+  { id: 1, name: 'Buy a book', amount: 20 },
+  { id: 2, name: 'Buy dog food', amount: 5 },
+  { id: 3, name: 'Book a flight ticket', amount: 225 },
+  { id: 4, name: 'Buy speakers', amount: 300 },
+  { id: 5, name: 'Book a concert ticket', amount: 50 },
+  { id: 6, name: 'Buy birthday present', amount: 100 },
+  { id: 7, name: 'Buy boots', amount: 400 },
+  { id: 7, name: 'Buy boots', amount: 400 }
+];
```
Comment (donni106) [inline screens/HomeScreen.js:19]: think  f empt lines please for readability and add one afterwards

File: screens/HomeScreen.js [hunk 3]
```diff
@@ -25,6 +24,10 @@ export default function HomeScreen() {
 
   const handleSubmitForm = () => {
     if (name !== '' && amount > 0) {
+      console.log('string?', amount);
+      // parseFloat(amount);
+      // amount.toString();
```
Comment (donni106) [inline screens/HomeScreen.js:29]: what do you expect from `toString`?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toString

File: screens/HomeScreen.js [hunk 4]
```diff
@@ -25,6 +24,10 @@ export default function HomeScreen() {
 
   const handleSubmitForm = () => {
     if (name !== '' && amount > 0) {
+      console.log('string?', amount);
+      // parseFloat(amount);
```
Comment (donni106) [inline screens/HomeScreen.js:28]: why was your idea to use `parseFloat`? what do you expect from it?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat

**Comment by donni106** [comment]
Your last commit says "Solving bug on total amount" but you have not solved it! This can be difficult for the history of the project.

File: screens/HomeScreen.js [hunk 5]
```diff
@@ -1,35 +1,56 @@
...
 
+const ALL_EXPENSES = [
+  { id: 1, name: 'Buy a book', amount: 20 },
+  { id: 2, name: 'Buy dog food', amount: 5 },
+  { id: 3, name: 'Book a flight ticket', amount: 225 },
+  { id: 4, name: 'Buy speakers', amount: 300 },
+  { id: 5, name: 'Book a concert ticket', amount: 50 },
+  { id: 6, name: 'Buy birthday present', amount: 100 },
+  { id: 7, name: 'Buy boots', amount: 400 },
+  { id: 7, name: 'Buy boots', amount: 1000 },
```
Comment (donni106) [inline screens/HomeScreen.js:19]: It is dummy data, right, but having the same "ID" is bad...IDs should be unique in any system.

File: screens/HomeScreen.js [hunk 6]
```diff
@@ -1,35 +1,56 @@
...
+  };
+
+  const handleName = (event) => {
+    console.log('Name ', event);
+    setName(event);
+  };
+
+  const handleAmount = (event) => {
+    console.log('Amount', event);
+    setAmount(parseInt(event, 10));
```
Comment (donni106) [inline screens/HomeScreen.js:45]: commenting this method oder line of code would be very nice for saying why this `parseInt` is necessary

Comment (SoleCincis) [reply screens/HomeScreen.js:45]: it's explained  in issue #4 !

File: screens/HomeScreen.js (hunk 5 above)
Comment (SoleCincis) [reply screens/HomeScreen.js:19]: right! this is a typo that coms from testing , last element should not be there ..

File: screens/HomeScreen.js [hunk 7]
```diff
@@ -43,6 +42,8 @@ export default function HomeScreen() {
   const handleAmount = (event) => {
     console.log('Amount', event);
     setAmount(parseInt(event, 10));
+    // transforming sting into integer value;
```
Comment (donni106) [inline screens/HomeScreen.js:45]: sting -> string

File: screens/HomeScreen.js [hunk 8]
```diff
@@ -43,6 +42,8 @@ export default function HomeScreen() {
...
+    // input field gives a sting but I need an integer in order to sum in total amount
```
Comment (donni106) [inline screens/HomeScreen.js:46]: yea! now everybody knows whats going on!

File: screens/HomeScreen.js (hunk 7 above)
Comment (SoleCincis) [reply screens/HomeScreen.js]: oops !

**Pull request merged**